sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncWithAnotherActivityViewModel.kt

```diff
@@ -299,4 +309,17 @@ class SyncWithAnotherActivityViewModel @Inject constructor(
     fun onUserAskedToSwitchAccount() {
         syncPixels.fireAskUserToSwitchAccount()
     }
+
+    private data class BarcodeContents(
+        /**
+         * The underlying code that was encoded in the barcode.
+         * It's possible this is different from the barcode string which might contain extra data
+         */
+        val underlyingCode: String,
+
+        /**
+         * The string that was encoded in the barcode.
+         */
+        val barcodeString: String,
```

Comment:
why barcodeString has `String` in the name while underlyingCode doesn't? both are Strings.
is underlyingCode also a barcode?